src/Module/Conversation/Timeline.php

```diff
@@ -0,0 +1,475 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2023, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Conversation;
+
+use Friendica\App;
+use Friendica\App\Mode;
+use Friendica\BaseModule;
+use Friendica\Content\Conversation\Collection\Timelines;
+use Friendica\Content\Conversation\Entity\Timeline as TimelineEntity;
+use Friendica\Core\Cache\Capability\ICanCache;
+use Friendica\Core\Cache\Enum\Duration;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\L10n;
+use Friendica\Core\PConfig\Capability\IManagePersonalConfigValues;
+use Friendica\Core\Renderer;
+use Friendica\Core\Session\Capability\IHandleUserSessions;
+use Friendica\Model\Contact;
+use Friendica\Model\User;
+use Friendica\Database\Database;
+use Friendica\Model\Item;
+use Friendica\Model\Post;
+use Friendica\Module\Response;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Util\Profiler;
+use Psr\Log\LoggerInterface;
+
+class Timeline extends BaseModule
+{
+	/** @var string */
+	protected static $selectedTab;
+	/** @var mixed */
+	protected static $min_id;
+	/** @var mixed */
+	protected static $max_id;
+	/** @var string */
+	protected static $accountTypeString;
+	/** @var int */
+	protected static $accountType;
+	/** @var int */
+	protected static $item_id;
+	/** @var int */
+	protected static $itemsPerPage;
+	/** @var bool */
+	protected static $no_sharer;
```

Comment:
If these are only called in object context, they don't need to be static properties anymore.

Reply:
It's done for community and channels, but not for the network. That class needs some work.

Reply:
Done